packages/pds/src/services/moderation/index.ts

```diff
@@ -215,13 +272,40 @@ export class ModerationService {
       .execute()
   }
 
-  async takedownRecord(info: { takedownId: number; uri: AtUri }) {
+  async takedownRecord(info: {
+    takedownId: number
+    uri: AtUri
+    blobCids?: CID[]
+  }) {
     await this.db.db
       .updateTable('record')
       .set({ takedownId: info.takedownId })
       .where('uri', '=', info.uri.toString())
       .where('takedownId', 'is', null)
       .executeTakeFirst()
+    if (info.blobCids?.length) {
+      await this.db.db
+        .updateTable('repo_blob')
+        .set({ takedownId: info.takedownId })
+        .where('recordUri', '=', info.uri.toString())
+        .where(
+          'cid',
+          'in',
+          info.blobCids.map((c) => c.toString()),
+        )
+        .where('takedownId', 'is', null)
+        .executeTakeFirst()
+      await Promise.all(
+        info.blobCids.map(async (cid) => {
+          await this.blobstore.quarantine(cid)
+          const paths = ImageUriBuilder.commonSignedUris.map((id) => {
+            const uri = this.imgUriBuilder.getCommonSignedUri(id, cid)
+            return uri.replace(this.imgUriBuilder.endpoint, '')
+          })
+          await this.imgInvalidator.invalidate(cid.toString(), paths)
```

Comment:
one thing i could see here is that an error thrown by either the blobstore or the invalidator would revert the txn & prevent the takedown of some record.

this is good for maintaining a consistent state but arguably bad if the goal is "stop showing this post with as little chance for hiccups as possible"